### PR TITLE
feat: alternative pypi registries

### PR DESCRIPTION
**Pull request by tdejager**
Enables the use of alternative PyPI registries. The definition of these can be specified per feature. There are certain rules for making unions out of these that are explained throughout the code.

This adds the options for:
* extra-index-urls
* index-urls
* find-links

Which follows the familiar pypi conventions. These are deserialized and passed **per** solve to uv.

### CODE REVIEW

**Comment by wolfv** [comment]
Here is some real-life environment.yml that uses the `--extra-index-url` argument of pip: https://github.com/zliucz/animate-your-word/blob/main/environment.yml. Would be nice to support that :)

**Comment by zachcp** [comment]
.... its getting close .....

**Comment by tdejager** [comment]
@vlad-ivanov-name https://github.com/prefix-dev/pixi/pull/1279 will add this. We can look at alternative authentication methods later :)

File: examples/pypi-custom-registry/pixi.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+authors = ["Tim de Jager <tim@prefix.dev>"]
+channels = ["conda-forge"]
+platforms = ["osx-arm64"]
+
+[dependencies]
+python = "3.12"
+
+[feature.alternative.pypi-options]
+index-url = "https://pypi.tuna.tsinghua.edu.cn/simple/"
+[feature.alternative.pypi-dependencies]
```
Comment (ruben-arts) [inline examples/pypi-custom-registry/pixi.toml:14]: Does this require a definition of that dependency here to be seen in the resolution?

**Review by ruben-arts — COMMENTED** [review]
Is there a way to test the find links feature?

**Comment by tdejager** [comment]
> Is there a way to test the find links feature?

There is an example using this.

File: examples/pypi-custom-registry/pixi.toml (hunk 1 above)
Comment (tdejager) [reply examples/pypi-custom-registry/pixi.toml:14]: Sorry I don't understand what you mean :)?

Comment (ruben-arts) [reply examples/pypi-custom-registry/pixi.toml:14]: Ah yeah, I meant to say, can't you just define the PyPi dependencies once?

Comment (tdejager) [reply examples/pypi-custom-registry/pixi.toml:14]: Sure, then it goes into the default feature

Comment (tdejager) [reply examples/pypi-custom-registry/pixi.toml:14]: At the time I made the example 'no-default' was not yet a thing but I could use that now 😀

**Comment by tdejager** [comment]
@ruben-arts I've added some documentation regarding to the feature.

**Review by ruben-arts — APPROVED** [review]
Awesome

**Comment by ianzone** [comment]
how to set find-links for specific package?

**Pull request merged**